internal/containerizedengine/update.go

```diff
@@ -102,30 +72,136 @@ func (c *baseClient) DoUpdate(ctx context.Context, opts clitypes.EngineInitOptio
 		}
 	}
 
-	// Gather information about the existing engine so we can recreate it
-	engine, err := c.GetEngine(ctx)
+	// Make sure we're safe to proceed
+	newMetadata, err := c.PreflightCheck(ctx, image)
 	if err != nil {
-		if err == ErrEngineNotPresent {
-			return errors.Wrap(err, "unable to find existing engine - please use init")
+		return err
+	}
+	// Grab current metadata for comparison purposes
+	if localMetadata != nil {
+		if localMetadata.Platform != newMetadata.Platform {
+			fmt.Fprintf(out, "\nNotice: you have switched to \"%s\".  Please refer to %s for update instructions.\n\n", newMetadata.Platform, c.GetReleaseNotesURL(imageName))
 		}
+	}
+
+	err = c.cclient.Install(ctx, image, containerd.WithInstallReplace, containerd.WithInstallPath("/usr"))
+	if err != nil {
 		return err
 	}
 
-	// TODO verify the image has changed and don't update if nothing has changed
+	return c.WriteRuntimeMetadata(ctx, "", newMetadata)
+}
+
+var defaultDockerRoot = "/var/lib/docker"
 
-	err = containerized.AtomicImageUpdate(ctx, engine, image, func() error {
-		ctx, cancel := context.WithTimeout(ctx, engineWaitTimeout)
-		defer cancel()
-		return c.waitForEngine(ctx, out, healthfn)
-	})
-	if err == nil && opts.Scope != "" {
-		var labels map[string]string
-		labels, err = engine.Labels(ctx)
+// GetCurrentRuntimeMetadata loads the current daemon runtime metadata information from the local host
+func (c *baseClient) GetCurrentRuntimeMetadata(_ context.Context, dockerRoot string) (*RuntimeMetadata, error) {
+	if dockerRoot == "" {
```

Comment:
Looks like the `dockerRoot` argument is only used in tests, but otherwise `/var/lib/docker` is hard-coded. 

- Should this argument be there if it's only used for testing?
- Is hardcoding `/var/lib/docker` a concern here? Using a different data-dir is a very common use-case, and we can obtain the correct data-dir from the daemon

Reply:
Lets revisit this in a follow up - agreed the dockerroot is going to be something we need to handle.